src/components/MAPI/clusters/CreateCluster/__tests__/index.tsx

```diff
@@ -0,0 +1,327 @@
+import ClusterCreate from '../';
+
+function getComponent(
+  props: React.ComponentPropsWithoutRef<typeof ClusterCreate>
+) {
+  const history = createMemoryHistory();
+  const auth = new TestOAuth2(history, true);
+
+  const Component = (p: typeof props) => (
+    <SWRConfig value={{ dedupingInterval: 0 }}>
+      <ClusterCreate {...p} />
+    </SWRConfig>
+  );
+
+  const state = {
+    ...preloginState,
+    main: {
+      ...preloginState.main,
+      info: {
+        ...preloginState.main.info,
+        general: {
+          ...preloginState.main.info.general,
+          availability_zones: {
+            default: 1,
+            max: 3,
+            zones: ['1', '2', '3'],
+          },
+          provider: Providers.AZURE,
+        },
+      },
+    },
+  };
+
+  return getComponentWithStore(
+    Component,
+    props,
+    state,
+    undefined,
+    history,
+    auth
+  );
+}
+
+jest.mock('react-router', () => ({
+  ...jest.requireActual('react-router'),
+  useRouteMatch: jest.fn().mockReturnValue({
+    url: '',
+    params: {
+      orgId: 'org1',
+    },
+  }),
+}));
+
+const generateUIDMockFn = jest.spyOn(MAPIUtils, 'generateUID');
+generateUIDMockFn.mockReturnValue(
+  capiv1alpha3Mocks.randomCluster1.metadata.name
+);
+
+describe('ClusterCreate', () => {
+  afterEach(() => {
+    cache.clear();
+  });
+
+  it('renders without crashing', () => {
+    render(getComponent({}));
+  });
+
+  it('can see the name that the resource will get', () => {
+    render(getComponent({}));
+
+    expect(screen.getByLabelText('Cluster name: j5y9m')).toBeInTheDocument();
+  });
+
+  it('can set the description', async () => {
+    const createClusterMockFn = jest.spyOn(CreateClusterUtils, 'createCluster');
+    createClusterMockFn.mockResolvedValue({
+      cluster: capiv1alpha3Mocks.randomCluster1,
+      providerCluster: capzv1alpha3Mocks.randomAzureCluster1,
+      controlPlaneNode: capzv1alpha3Mocks.randomAzureMachine1,
+    });
+
+    nock(window.config.mapiEndpoint)
+      .get('/apis/release.giantswarm.io/v1alpha1/releases/')
+      .reply(StatusCodes.Ok, releasev1alpha1Mocks.releasesList);
+
+    render(getComponent({}));
+
+    fireEvent.change(screen.getByLabelText('Description'), {
+      target: { value: 'A new description' },
+    });
+
+    const createButton = screen.getByRole('button', { name: 'Create cluster' });
+
+    await waitFor(() => {
+      expect(createButton).not.toBeDisabled();
+    });
+
+    fireEvent.click(createButton);
+
+    expect(createClusterMockFn).toHaveBeenCalledWith(
+      expect.any(Object),
+      expect.any(Object),
+      expect.objectContaining({
+        cluster: expect.objectContaining({
+          metadata: expect.objectContaining({
+            annotations: expect.objectContaining({
+              'cluster.giantswarm.io/description': 'A new description',
+            }),
+          }),
+        }),
+      })
+    );
+
+    // TODO(axbarsan): Replace with assertion utility.
+    expect(
```

Comment:
This needs an assertion utility from https://github.com/giantswarm/happa/pull/2551

I will replace this code with that once the other PR gets merged